docs/source/concepts/architectural-model.rst

```diff
@@ -0,0 +1,284 @@
+This is because the endpoint cannot know at the ``LeftViewEvent`` that there will be another ``EnteredViewEvent``.
+
+In contrast, `P2` need not output the ``EnteredViewEvent``\ s and ``LeftViewEvent``\ s at all in this example because `P2` hosts `A` on both domains.
+
+
+Relaxing the total ordering
+***************************
+
+The virtual shared ledger and the transaction service put all transactions into a linear sequence.
+Establishing such an order at all endpoints incurs synchronization, which hinders scalability.
+Different endpoints are therefore allowed to output events in different orders if these events do not causally depend on each other from the party's perspective.
+This section defines :ref:`causality <ledger-causality-party>` and illustrates the allowed and disallowed reorderings of commits by :ref:`examples <ledger-causality-examples>`.
+
+.. _ledger-causality-party:
+
+Causality
+=========
+
+Causality is defined on the commits in a party's projection of the virtual shared ledger.
+
+Definition »synchronizes with«
+  An action `act`\ :sub:`1` **synchronizes with** an action `act`\ :sub:`2` for party `A` if one of the following holds:
+
+  * **Create-use:**
+    `act`\ :sub:`1` creates the contract `c` and `act`\ :sub:`2` uses the contract, i.e., `act`\ :sub:`2` is a **Fetch** or an **Exercise** on `c` and `A` is a stakeholder of `c` and an informee of `act`\ :sub:`2`.
+ 
+  * **Use-use:**
+    `act`\ :sub:`1` and `act`\ :sub:`2` both use a contract `c` and `A` is a stakeholder of `c` and an informee of both actions.
+
+  * **Contract keys**:
+    `act`\ :sub:`1` and `act`\ :sub:`2` both are actions on a key `k` and `A` is a maintainer of `k`.
+
+Definition »causal order«
+  A commit `C`\ :sub:`2` **causally depends on** another commit `C`\ :sub:`1` for party `A` if all of the following holds:
+  * `C`\ :sub:`1` precedes `C`\ :sub:`2` on the virtual shared ledger.
+  * They contain actions `act`\ :sub:`2` and `act`\ :sub:`1` such that `act`\ :sub:`1` synchronizes with `act`\ :sub:`2` for party `A`.
+  
+  A commit `C`\ :sub:`1` is **causally ordered** before `C`\ :sub:`2` for `A` if `C`\ :sub:`2` causally depends on `C`\ :sub:`1` for `A` or if there is another commit `C'` in `A`\ 's projection such that `C`\ :sub:`1` is causally ordered before `C'` for `A` and `C'` is causally ordered before `C`\ :sub:`2` for `A`.
+
+.. note::
+   The synchronizes with relationship is not symmetric.
+   If `act`\ :sub:`1` uses a contract and `act`\ :sub:`2` creates it, then `act`\ :sub:`1` does not synchronize with `act` :sub:`2`.
+   This is in line with how the concept is used in programming languages: writes to a memory location synchronize with reads, but reads do not synchronize with writes.
+   This asymmetry does not matter for the causal ordering in a consistent ledger as **Create**\ s precede usages on such a ledger.
+  
+.. _ledger-api-guarantees:
+  
+Ledger API ordering guarantees
```

Comment:
The way you've set this doc up makes this last part, which is what people actually care about, very complicated.

- VSLs are a DAG
- A Party projection is a subgraph (and thus a DAG), sub-ledger, subVSL
- A Party projection is a subgraph (and thus a DAG), sub-ledger, subVSL
- On any such graph you can define the transitive closure of the ordering, which is causality order
- Ledger API guarantees to return transactions according to causality order for the transaction graph available for thre party on the Participant Node